File: chambers/src/main/java/vip/qoriginal/quantumplugin/chambers/ChamberRunResult.java
package vip.qoriginal.quantumplugin.chambers;

import vip.qoriginal.quantumplugin.registration.MinecraftRegistrationTest;

public record ChamberRunResult(
        MinecraftRegistrationTest.Session registrationSession,
        FinishReason reason,
        int completedChambers,
        int totalChambers
) {
    public boolean passed() {
        return reason == FinishReason.PASSED;
    }

    public enum FinishReason {
        PASSED,
        TIMED_OUT,
        CANCELLED,
        QUIT,
        PLUGIN_DISABLED
    }
}

File: chambers/src/main/java/vip/qoriginal/quantumplugin/chambers/data/ChamberRunResult.kt
package vip.qoriginal.quantumplugin.chambers.data

import vip.qoriginal.quantumplugin.registration.MinecraftRegistrationTest

data class ChamberRunResult(
	val registrationSession: MinecraftRegistrationTest.Session?,
	val reason: FinishReason,
	val completedChambers: Int,
	val totalChambers: Int,
) {
    val passed: Boolean
        get() = reason == FinishReason.PASSED

    enum class FinishReason {
        PASSED,
        TIMED_OUT,
        CANCELLED,
        QUIT,
        PLUGIN_DISABLED,
    }
}